docs/installation/database/passing-configuration.rst

```diff
@@ -0,0 +1,430 @@
+   to get an error by requesting an environment that does not exist.  Instead,
+   an all-defaults environment with the requested name will be dynamically
+   created.  The only exception is when a specific environment is not
+   requested.  In this case, the ``default`` environment is only used if it is
+   either defined in the active configuration file or has previously been
+   explicitly requested from the same :py:class:`ODCConfig` object.
+
+3. Although environment variable overrides are bypassed for configured
+   environments by passing in explicit configuration, reading from environment
+   variables to dynamically create new environments is still supported.
+
+4b. Environment Variable Overrides and Environment Aliases
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To avoid troublesome and unpredictable corner cases, aliases can only be
+defined in raw configuration or in config files - they cannot be defined
+through environment variables.
+
+i.e. defining ``ODC_ENV2_ALIAS=env1`` does NOT create an ``env2`` alias to the ``env1``
+environment.
+
+A configuration file may define an environment which is an alias to an environment that is to be loaded
+dynamically and is NOT defined in the configuration file.
+
+Aliases (created in raw config or a config file) **ARE** honoured when interpreting environment variables.
+
+E.g.  Given config file:
+
+.. code-block::yaml
+
+     default:
+          alias: main
+     common:
+          alias: main
+     main:
+          index_driver: postgis
+          db_url: postgresql://uid:pwd@server.domain:5432/main
+
+The "main" environment url can be over-ridden with **ANY** of the following environment variables:
+
+.. code-block::bash
+
+   $ODC_DEFAULT_DB_URL
+   $ODC_COMMON_DB_URL
+   $ODC_MAIN_DB_URL
+
+The environment variable using the canonical environment name (``$ODC_MAIN_DB_URL`` in this case) always
+takes precedence if it set. If more than one alias environment name is used (e.g. if both ``$ODC_DEFAULT_DB_URL``
+**AND** ``$ODC_COMMON_DB_URL`` exist) then only one will be read and the implementation makes no guarantees
+about which.  Therefore canonical environment names are strongly recommended for environment variable names where
+possible.
+
+4c. Deprecated Legacy Environment Variables
++++++++++++++++++++++++++++++++++++++++++++
+
+Some legacy environment variable names are also read for backwards
+compatibility reasons, however they may not work as expected where more than
+one ODC environment is in use and will generate a deprecation warning if they
+are read from.  The preferred new environment variable name will be included in
+the text of the deprecation warning.
+
+Most notably the old database connection environment variables:
+
+.. code-block::bash
+
+   $DB_DATABASE
+   $DB_HOSTNAME
+   $DB_PORT
+   $DB_USERNAME
+   $DB_PASSWORD
+
+are strongly deprecated as they will be applied to ALL environments, which is probably not what you intended.
```

Comment:
I think things are either deprecated or not, so "strongly" can probably be omitted.